src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,444 @@
+        WT_ASSERT(session, lr_fs->state == WTI_LIVE_RESTORE_STATE_LOG_COPY);
+        break;
+    case WTI_LIVE_RESTORE_STATE_CLEAN_UP:
+        WT_ASSERT(session, lr_fs->state == WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION);
+        break;
+    case WTI_LIVE_RESTORE_STATE_COMPLETE:
+        WT_ASSERT(session, lr_fs->state == WTI_LIVE_RESTORE_STATE_CLEAN_UP);
+        break;
+    }
+
+    WT_DECL_ITEM(state_file_name);
+    WT_ERR(__wt_scr_alloc(session, 0, &state_file_name));
+
+    WT_ERR(__wt_filename_construct(session, lr_fs->destination.home, WTI_LIVE_RESTORE_STATE_FILE,
+      UINTMAX_MAX, UINT32_MAX, state_file_name));
+    WT_ERR(lr_fs->os_file_system->fs_exist(lr_fs->os_file_system, (WT_SESSION *)session,
+      (char *)state_file_name->data, &state_file_exists));
+
+    /*
+     * The state file is either already present or created on live restore initialization. If it's
+     * not present we've called set state too early.
+     */
+    WT_ASSERT_ALWAYS(session, state_file_exists, "State file doesn't exist!");
+
+    char state_to_write[128];
+    __live_restore_state_to_string(new_state, state_to_write);
+    WT_ERR(lr_fs->os_file_system->fs_open_file(lr_fs->os_file_system, (WT_SESSION *)session,
+      (char *)state_file_name->data, WT_FS_OPEN_FILE_TYPE_REGULAR, WT_FS_OPEN_EXCLUSIVE, &fh));
+    WT_ERR(fh->fh_write(fh, (WT_SESSION *)session, 0, 128, state_to_write));
+
+    lr_fs->state = new_state;
+    __live_restore_set_public_state(session, new_state);
+
+err:
+    __wt_writeunlock(session, &lr_fs->state_lock);
+
+    if (fh != NULL)
+        WT_TRET(fh->close(fh, &session->iface));
+
+    __wt_scr_free(session, &state_file_name);
+
+    return (ret);
+}
+
+/*
+ * __wti_live_restore_init_state --
+ *     Initialize the live restore state. Read the state from file if it exists, otherwise we start
+ *     in the log copy state and need to create the file on disk.
+ */
+int
+__wti_live_restore_init_state(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs)
+{
+    WT_DECL_RET;
+    WT_FILE_HANDLE *fh = NULL;
+
+    __wt_writelock(session, &lr_fs->state_lock);
+
+    WT_ASSERT_ALWAYS(session, lr_fs->state == WTI_LIVE_RESTORE_STATE_NONE,
+      "Attempting to initialize already initialized state!");
+
+    WT_DECL_ITEM(state_file_name);
+    WTI_LIVE_RESTORE_STATE state;
+    WT_RET(__live_restore_get_state_from_file(
+      session, lr_fs->os_file_system, lr_fs->destination.home, &state));
+
+    if (state != WTI_LIVE_RESTORE_STATE_NONE) {
+        lr_fs->state = state;
+    } else {
+        /*
+         * The state file doesn't exist which means we're starting a brand new live restore. Create
+         * the state file in the log copy state.
+         */
+        char state_to_write[128];
+        __live_restore_state_to_string(WTI_LIVE_RESTORE_STATE_LOG_COPY, state_to_write);
+
+        WT_RET(__wt_scr_alloc(session, 0, &state_file_name));
+        WT_ERR(__wt_filename_construct(session, lr_fs->destination.home,
```

Comment:
Having two state file creation paths seems odd, can they really not be wound together?